SmartDeviceLink/SDLError.m

```diff
@@ -25,6 +26,44 @@
 
 @implementation NSError (SDLErrors)
 
+#pragma mark - SDLEncryptionLifecycleManager
++ (NSError *)sdl_encryption_lifecycle_notReadyError {
+    NSDictionary<NSString *, NSString *> *userInfo = @{
+                                                       NSLocalizedDescriptionKey: NSLocalizedString(@"Encryption Lifecycle manager not ready", nil),
+                                                       NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The SDL library is not finished setting up the connection, please wait until the encryption lifecycleState is SDLEncryptionLifecycleStateReady", nil),
+                                                       NSLocalizedRecoverySuggestionErrorKey: NSLocalizedString(@"Make sure HMI is not NONE and at least one RPC requires encryption in permissions?", nil)
+                                                       };
+    
+    return [NSError errorWithDomain:SDLErrorDomainEncryptionLifecycleManager
+                               code:SDLEncryptionLifecycleManagerErrorNotConnected
+                           userInfo:userInfo];
+}
+
++ (NSError *)sdl_encryption_lifecycle_encryption_off {
+    NSDictionary<NSString *, NSString *> *userInfo = @{
+                                                       NSLocalizedDescriptionKey: NSLocalizedString(@"Encryption Lifecycle received a ACK with encryption bit = 0", nil),
+                                                       NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The SDL library received ACK with encryption = OFF.", nil),
+                                                       NSLocalizedRecoverySuggestionErrorKey: NSLocalizedString(@"Make sure you are on a supported remote head unit with proper policies and your app id is approved.", nil)
+                                                       };
+    
+    return [NSError errorWithDomain:SDLErrorDomainEncryptionLifecycleManager
+                               code:SDLEncryptionLifecycleManagerErrorEncryptionOff
+                           userInfo:userInfo];
+}
+
++ (NSError *)sdl_encryption_lifecycle_nack {
+    NSDictionary<NSString *, NSString *> *userInfo = @{
+                                                       NSLocalizedDescriptionKey: NSLocalizedString(@"Encryption Lifecycle received a negative acknowledgement", nil),
+                                                       NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The remote head unit sent a nACK.  Encryption service failed to start due to nACK.", nil),
+                                                       NSLocalizedRecoverySuggestionErrorKey: NSLocalizedString(@"Make sure your certificates are valid.", nil)
+                                                       };
+    
+    return [NSError errorWithDomain:SDLErrorDomainEncryptionLifecycleManager
+                               code:SDLEncryptionLifecycleManagerErrorNACK
```

Comment:
Both are acceptable and mean the same thing; NAK is not a typo and commonly used. A quick google search will show you this. Please try to focus on fixing issues the project maintainer finds so this feature can be included in the release and we don't have to pull it out.

Reply:
Makes sense sorry about that, mainly wanted to mention there is inconsistency in the code between `NACK` and `NAK`.

Reply:
Inconsistencies aren't good either. Our [protocol spec defines them as NAK](https://github.com/smartdevicelink/protocol_spec#23-frame-header-fields) so that should be the standard we use in the project. Not sure when `SDLFrameInfoStartServiceNACK` was introduced, but I would say it is the outlier, but the pay off has to be worth the effort (deprecation, breaking changes, etc).

Reply:
Sounds good, good thing we have a knack for this